test/transaction.cpp

```diff
@@ -152,6 +151,9 @@ void doTransactionTests(json_spirit::mValue& _v, bool _fillin)
 				if (!txFromFields.signature().isValid())
 					BOOST_THROW_EXCEPTION(Exception() << errinfo_comment("transaction from RLP signature is invalid") );
 
+				//cause Address is length20 array, when trying to create address from sting of another length, field "to" would be diffrent from RLP encoded Address
+				BOOST_CHECK_MESSAGE(Address(tObj["to"].get_str()) == txFromFields.receiveAddress(), "seems that transaction 'to' address has wrong format");
```

Comment:
If I put in a sting which is longer than 20, but looks like 0xffff...fff . You still get a valid address.
But also you do not need this line at all. When creating the Transaction from an RLP this line gets executed:
m_receiveAddress = rlp[field = 3].toHash<Address>();
In toHash an exception is thrown if the size of the input data does not match the size of Address. Therefore you could never reach that line in that case.